### ISSUE TITLE
Attribute `@` incorrectly marked as invalid

### ISSUE THREAD

**Issue by somebody1234** [vuejs/eslint-plugin-vue#2284]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.47.0
- **eslint-plugin-vue version:** 9.17.0
- **Node version:** 18.17.1
- **Operating System:** Linux

**Please show your full configuration:**
```js
/* eslint-env node */
require('@rushstack/eslint-patch/modern-module-resolution')

module.exports = {
  root: true,
  extends: [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    '@vue/eslint-config-typescript',
    '@vue/eslint-config-prettier/skip-formatting',
  ],
  parserOptions: {
    ecmaVersion: 'latest',
  },
}
```

**What did you do?**
```vue
    <div @="events"></div>
```

**What did you expect to happen?**
No errors

**What actually happened?**
```
Attribute name @ is not valid.  eslint(vue/valid-attribute-name)
```

**Repository to reproduce this issue**
https://github.com/somebody1234/vue-invalid-attribute

**Comment by somebody1234** [comment]
duplicate of #2091. unfortunately this behavior is incorrect - likely unspecified by the spec
